core/sr-io/without_std.rs

```diff
@@ -43,41 +51,214 @@ pub extern fn oom(_: ::core::alloc::Layout) -> ! {
 	static OOM_MSG: &str = "Runtime memory exhausted. Aborting";
 
 	unsafe {
-		ext_print_utf8(OOM_MSG.as_ptr(), OOM_MSG.len() as u32);
+		extern_functions_host_impl::ext_print_utf8(OOM_MSG.as_ptr(), OOM_MSG.len() as u32);
 		intrinsics::abort();
 	}
 }
 
+/// A function which implementation can be exchanged.
+///
+/// Internally this works by swapping function pointers.
+pub struct ExchangeableFunction<T>(T);
+
+impl<T> ExchangeableFunction<T> {
+	/// Replace the implementation with `new_impl`.
+	pub fn replace_implementation(self: &'static mut Self, new_impl: T)  -> RestoreImplementation<T> {
```

Comment:
where can you actually get a `&'static mut`?

Reply:
From a `static mut Type` .. :sweat_smile: